src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteNwContext.cs

```diff
@@ -0,0 +1,761 @@
+        {
+            Interop.NetworkFramework.Tls.SetTlsOptions(ConnectionHandle, GCHandle.ToIntPtr(_thisHandle), options);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+
+                Shutdown();
+
+                // Wait for the transport read task to complete
+                if (_transportReadTask is Task transportTask)
+                {
+                    try
+                    {
+                        transportTask.Wait();
+                        transportTask.Dispose();
+                    }
+                    catch
+                    {
+                        // Ignore exceptions from the transport task
+                    }
+                }
+
+                if (_pendingAppReceiveBufferFillTask is Task t)
+                {
+                    try
+                    {
+                        t.Wait();
+                        t.Dispose();
+                    }
+                    catch
+                    {
+                        // Ignore exceptions from the pending task, as we are disposing
+                        // and it may have been cancelled or completed with an exception.
+                    }
+                }
+
+                // wait for callback signalling connection has been truly closed.
+                _connectionClosedTcs.Task.GetAwaiter().GetResult();
+                // Complete all pending operations with ObjectDisposedException
+                var disposedException = new ObjectDisposedException(nameof(SafeDeleteNwContext));
+
+                _appReceiveBufferTcs.TrySetException(disposedException);
+                _transportReadTcs.TrySetException(disposedException);
+                _handshakeCompletionSource.TrySetException(disposedException);
+
+                // Complete any pending writes with disposed exception
+                TaskCompletionSource? writeCompletion = _currentWriteCompletionSource;
+                if (writeCompletion != null)
+                {
+                    _currentWriteCompletionSource = null;
+                    writeCompletion.TrySetException(new ObjectDisposedException(nameof(SafeDeleteNwContext)));
+                }
+
+                ConnectionHandle.Dispose();
+                _framerHandle?.Dispose();
+                _peerCertChainHandle?.Dispose();
+                _appReceiveBuffer.Dispose();
+                _shutdownCts?.Dispose();
+            }
+            base.Dispose(disposing);
+        }
+
+        private static SafeDeleteNwContext ResolveThisHandle(IntPtr thisHandle)
+        {
+            GCHandle handle = GCHandle.FromIntPtr(thisHandle);
+            if (!handle.IsAllocated)
+            {
+                throw new ObjectDisposedException(nameof(SafeDeleteNwContext), "Handle has been freed");
+            }
+            return (SafeDeleteNwContext)handle.Target!;
+        }
```

Comment:
This should not happen, if the thisHandle is not valid it means we have lifetime management problem, we should never get to a place where the IntPtr GCHandle we pass to native code is not valid and alive